flink-core/src/main/java/org/apache/flink/api/common/resources/Resource.java

```diff
@@ -51,50 +52,51 @@ protected Resource(String name, BigDecimal value) {
         this.value = value;
     }
 
-    public Resource merge(Resource other) {
+    public T merge(T other) {
         checkNotNull(other, "Cannot merge with null resources");
         checkArgument(getClass() == other.getClass(), "Merge with different resource type");
-        checkArgument(name.equals(other.name), "Merge with different resource name");
+        checkArgument(name.equals(other.getName()), "Merge with different resource name");
 
-        return create(value.add(other.value));
+        return create(value.add(other.getValue()));
     }
 
-    public Resource subtract(Resource other) {
+    public T subtract(T other) {
         checkNotNull(other, "Cannot subtract null resources");
         checkArgument(getClass() == other.getClass(), "Minus with different resource type");
-        checkArgument(name.equals(other.name), "Minus with different resource name");
+        checkArgument(name.equals(other.getName()), "Minus with different resource name");
         checkArgument(
-                value.compareTo(other.value) >= 0,
+                value.compareTo(other.getValue()) >= 0,
                 "Try to subtract a larger resource from this one.");
 
-        return create(value.subtract(other.value));
+        return create(value.subtract(other.getValue()));
     }
 
-    public Resource multiply(BigDecimal multiplier) {
+    public T multiply(BigDecimal multiplier) {
         return create(value.multiply(multiplier));
     }
 
-    public Resource multiply(int multiplier) {
+    public T multiply(int multiplier) {
         return multiply(BigDecimal.valueOf(multiplier));
     }
 
-    public Resource divide(BigDecimal by) {
+    public T divide(BigDecimal by) {
         return create(value.divide(by, 16, RoundingMode.DOWN));
     }
 
-    public Resource divide(int by) {
+    public T divide(int by) {
         return divide(BigDecimal.valueOf(by));
     }
 
     @Override
+    @SuppressWarnings("unchecked")
```

Comment:
It would be nice to keep the scope of `@SuppressWarnings` as small as possible. In this case, the single statement `T other = (T) o;` should be enough.